File: 2540-minimum-common-value/solution.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int getCommon(vector<int>& nums1, vector<int>& nums2) {
        int i = 0; // Pointer for nums1
        int j = 0; // Pointer for nums2
        int n = nums1.size();
        int m = nums2.size();

        // Traverse both arrays using two pointers
        while (i < n && j < m) {
            if (nums1[i] == nums2[j]) {
                // Since arrays are sorted, the first match found is the minimum
                return nums1[i];
            } else if (nums1[i] < nums2[j]) {
                // Increment pointer of the array with the smaller value
                i++;
            } else {
                j++;
            }
        }

        // No common element found
        return -1;
    }
};

File: 2540-minimum-common-value/solution.py
class Solution:
    def getCommon(self, nums1: List[int], nums2: List[int]) -> int:
        # Initialize two pointers for both arrays
        i, j = 0, 0
        n, m = len(nums1), len(nums2)
        
        # Iterate while pointers are within the bounds of both arrays
        while i < n and j < m:
            if nums1[i] == nums2[j]:
                # Found the smallest common element
                return nums1[i]
            elif nums1[i] < nums2[j]:
                # Move the pointer of the smaller element forward
                i += 1
            else:
                # Move the pointer of the smaller element forward
                j += 1
                
        # If the loop finishes without returning, no common element exists
        return -1